leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -160,7 +154,7 @@ public LeadershipToken blockOnBecomingLeader() throws InterruptedException {
 
     private void proposeLeadershipOrWaitForBackoff(LeadershipState currentState)
             throws InterruptedException {
-        if (pingLeader()) {
+        if (pingLeader(currentState.greatestLearnedValue())) {
```

Comment:
Do we need this piece? I think it's safe, but at least right now `knowledge.getGreatestLearnedValue()` is cheap, and I'd prefer not to keep Paxos constructs in scope where not strictly needed

Reply:
snippet of next refactor came through, basically, I want the construct `PingableLeader.pingLeaderWithUuid`. If I have an external pinger, it just needs to know about all uuids, and can coalesce the uuid resolution calls. Otherwise there'd need to be an intermediate layer that can retrieve knowledge (client-leader specific) and then pass that onto the autobatching pinger. I want most of the logic to live inside `PaxosLeaderElectionService` so I don't have to duplicate in both autobatch and non-autobatch case non trivial elements.

Plus it's weird that after we get the leadership state, `pingLeader` is the only place that rerequests latest knowledge, we should use the state we've just learnt.

Reply:
Agree it's inconsistent with the others (but I did entertain the hypothesis that it was inconsistent for a specific reason - even though I could find no such reason).

Yep, this seems reasonable if we need it.